### PR TITLE
Cleaned up the localization code

### PR DESCRIPTION
**Pull request by AlexandreSanscartier**

### CODE REVIEW

File: CDTS Blazor/Model/BaseComponent.cs [hunk 1]
```diff
@@ -2,34 +2,42 @@
...
         [Parameter]
         public string LanguageCode { get; set; }
 
-        [System.Diagnostics.CodeAnalysis.SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1600:Elements should be documented", Justification = "Testing purposes")]
-        [System.Diagnostics.CodeAnalysis.SuppressMessage("Design", "CA1056:Uri properties should not be strings", Justification = "Testing purposes")]
-        public string LangBaseUri => this.navigationManager.BaseUri + this.LanguageCode;
+        /// <summary>
+        /// Gets the base uri for the language.
+        /// </summary>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Design", "CA1056:Uri properties should not be strings", Justification = "In this case we want this tu be a string")]
```
Comment (tc-tibo) [inline CDTS Blazor/Model/BaseComponent.cs:31]:
```suggestion
        [System.Diagnostics.CodeAnalysis.SuppressMessage("Design", "CA1056:Uri properties should not be strings", Justification = "In this case we want this to be a string")]
```

File: CDTS Blazor/Utilities/SessionHelper.cs [hunk 2]
```diff
@@ -1,41 +1,76 @@
...
-    [System.Diagnostics.CodeAnalysis.SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1600:Elements should be documented", Justification = "Testing purposes")]
-    public class SessionHelper
+    /// <summary>
+    /// The application's session helper.
+    /// </summary>
+    public sealed class SessionHelper
     {
-        [System.Diagnostics.CodeAnalysis.SuppressMessage("Design", "CA1062:Validate arguments of public methods", Justification = "Testing purposes")]
+        // TODO: This should be in a config file
+        private const string DEFAULTLANGUAGE = "en";
```
Comment (tc-tibo) [inline CDTS Blazor/Utilities/SessionHelper.cs:12]:
```suggestion
        private const string DefaultLanguage= "en";
```
https://docs.microsoft.com/en-us/previous-versions/dotnet/netframework-3.5/ms229043(v=vs.90)?redirectedfrom=MSDN suggests PascalCasing for "Read-only static field"

**Comment by TCPrimedPaul** [comment]
Nothing to add, great work! I really like that you added short descriptions for every class's responsibility/role.

File: CDTS Blazor/Utilities/CustomRequestCultureProvider.cs [hunk 3]
```diff
@@ -5,7 +5,9 @@
     using Microsoft.AspNetCore.Http;
     using Microsoft.AspNetCore.Localization;
 
-    [System.Diagnostics.CodeAnalysis.SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1600:Elements should be documented", Justification = "Testing purposes")]
+    /// <summary>
+    /// The application's request culture provider for managing multiple languages (en and fr in this case).
+    /// </summary>
     public class CustomRequestCultureProvider : RequestCultureProvider
     {
         private readonly string englishCulture = "en-CA";
```
Comment (tc-chris) [inline CDTS Blazor/Utilities/CustomRequestCultureProvider.cs:13]: Why en-CA and not en? I see french as fr and not fr-CA.

File: CDTS Blazor/Utilities/SessionHelper.cs [hunk 4]
```diff
@@ -1,41 +1,76 @@
...
-            if (!context.Request.Path.HasValue)
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            if (SessionHelper.IsContextPathEmpty(context))
             {
-                return "en";
+                return DEFAULTLANGUAGE;
```
Comment (tc-tibo) [inline CDTS Blazor/Utilities/SessionHelper.cs:29]:
```suggestion
                return DefaultLanguage;
```

File: CDTS Blazor/Utilities/SessionHelper.cs [hunk 5]
```diff
@@ -1,41 +1,76 @@
...
+        /// Gets the language from the uri path.
+        /// </summary>
+        /// <param name="path">the uri path.</param>
+        /// <returns>the language code.</returns>
         public static string GetLanguageFromPath(string path)
         {
             if (string.IsNullOrEmpty(path))
             {
-                return "en";
+                return DEFAULTLANGUAGE;
```
Comment (tc-tibo) [inline CDTS Blazor/Utilities/SessionHelper.cs:49]:
```suggestion
                return DefaultLanguage;
```

File: CDTS Blazor/Utilities/SessionHelper.cs [hunk 6]
```diff
@@ -1,41 +1,76 @@
...
             }
 
-            var items = path.Split('/');
-            if (items.Length < 2)
+            var pathSubDirectories = path.Split('/');
+            if (SessionHelper.DoesURLPathContainOnlyOneSubdirectory(pathSubDirectories))
             {
-                return "en";
+                return DEFAULTLANGUAGE;
```
Comment (tc-tibo) [inline CDTS Blazor/Utilities/SessionHelper.cs:55]:
```suggestion
                return DefaultLanguage;
```

File: CDTS Blazor/Utilities/SessionHelper.cs [hunk 7]
```diff
@@ -1,41 +1,76 @@
...
             }
 
-            if (string.IsNullOrEmpty(items[1]))
+            if (SessionHelper.IsSecondPathInURLNullOrEmpty(pathSubDirectories))
             {
-                return "en";
+                return DEFAULTLANGUAGE;
```
Comment (tc-tibo) [inline CDTS Blazor/Utilities/SessionHelper.cs:60]:
```suggestion
                return DefaultLanguage;
```

**Pull request merged**